lib/analyser.js

```diff
@@ -273,8 +297,10 @@ module.exports = {
           let toRemove = Array.isArray(whitelisted) ? whitelisted : [component];
 
           toRemove.forEach(item => {
-            let unusedIndex = this.unusedComponents.indexOf(item);
-            this.unusedComponents.splice(unusedIndex, 1);
+            let unusedValue = Object.values(this.unusedComponents).find(
+              component => item === component.name
```

Comment:
The whitelist still takes a flat list of strings for component names, so it was slightly different. Not sure if that's good, and maybe the whitelist should take a list of `keys`, where a key for a addon component to be ignored would be "[addon-name] component-name"?

Reply:
Prefixing addon's components sounds reasonable. At least it will save us from name clashes and it would be more verbose and explanatory. It shouldn't complicate this "check" too much, right?